das/daser.go

```diff
@@ -75,6 +118,58 @@ func (d *DASer) sampling(ctx context.Context, sub header.Subscription) {
 			continue
 		}
 
+		// If the next header coming through gossipsub is not adjacent
+		// to our last DASed header, kick off routine to DAS all headers
+		// between last DASed header and h. This situation could occur
+		// either on start or due to network latency/disconnection.
+		if h.Height > (height + 1) {
+			// DAS headers between last DASed height up to the current
+			// header
+			go d.catchUp(ctx, height, h.Height-1)
+		}
+
+		startTime := time.Now()
+
+		err = d.da.SharesAvailable(ctx, h.DAH)
+		if err != nil {
+			if err == context.Canceled {
+				return
+			}
+			log.Errorw("sampling failed", "height", h.Height, "hash", h.Hash(),
+				"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "err", err)
+			// continue sampling
+		}
+
+		sampleTime := time.Since(startTime)
+		log.Infow("sampling successful", "height", h.Height, "hash", h.Hash(),
+			"square width", len(h.DAH.RowsRoots), "finished (s)", sampleTime.Seconds())
+
+		height = h.Height
+	}
+}
+
+// catchUp starts sampling headers from the last known
+// checkpoint (latest/highest DASed header) `from`, and breaks the loop
+// once network head `to` is reached. (from:to]
```

Comment:
This comment refers to a last known checkpoint, but really it's catching up from `from`, which could be anything.

Reply:
resolved?